lib/public_activity/utility/store_controller.rb

```diff
@@ -1,24 +1,12 @@
 module PublicActivity
-  # @private
-  @@controllers = Hash.new
-  # Lambda called after the thread is destroyed.
-  Finalizer = lambda { |id|
-    @@controllers.delete id
-  }
-
-  class << self
-    # Setter for remembering controller instance
-    def set_controller(controller)
-      unless @@controllers.has_key?(Thread.current.object_id)
-        ObjectSpace.define_finalizer Thread.current, Finalizer
-      end if RUBY_VERSION != "1.9.3"
-      @@controllers[Thread.current.object_id] = controller
-    end
+  # Setter for remembering controller instance
+  def self.set_controller(controller)
```

Comment:
It's all style preference. Are you going somewhere with those questions or do you just want to learn?